### DISCUSSION TITLE
How to control the number of CPU cores used by POT?

### DISCUSSION THREAD

**Discussion by gangcai** [PythonOT/POT#385] [General]
## Describe the bug

It seems that for some functions of POT, all the CPU cores available for the machine will be used. How can I control the number of CPU cores to be used?  For example, if the computer has 20 CPU cores, all of them will be used by POT, how can I control only part of them (such as 10 CPU cores) being used?

**Comment by rflamary** [comment]
Functions such as emd/emd2 have a parameter `numThreads` that allows to select the number of CPU used by the solver.

But note that most solver actually use numpy that use multiple processors on marix multiplucation operation automatically. For those functiuosn you need to tell NUMPY how many cpu you want to use. On linux you can usually set them with the environment variable 
```
export OPENBLAS_NUM_THREADS = 1
```
